system/security/shared/caching.cpp

```diff
@@ -672,16 +680,47 @@ bool CPermissionsCache::queryPermsManagedFileScope(ISecUser& sec_user, const cha
 
 SecAccessFlags CPermissionsCache::queryDefaultPermission(ISecUser& user)
 {
-    if (m_defaultPermission == SecAccess_Unknown)
+
+    if (m_useLegacyDefaultFileScopePermissionCache)
+    {
+        if (m_defaultPermission == SecAccess_Unknown)
+        {
+            if (m_secMgr)
+                m_defaultPermission = m_secMgr->queryDefaultPermission(user);
+            else
+                m_defaultPermission = SecAccess_None;
```

Comment:
I changed this to SecAccess_Full. If no security manager then full access. I'm pretty sure the code essentially prevents this from being called if there is no security manager, so it may be moot, but correct none the less.